user_sync/rules.py

```diff
@@ -357,25 +360,39 @@ def create_remove_groups_callback(user_key):
         
         for organization_name, dashboard_connector in dashboard_connectors.get_accessor_connectors().iteritems():
             organization_info = self.get_organization_info(organization_name)
-            mapped_groups = organization_info.get_mapped_groups()
-            if (len(mapped_groups) == 0):
+            target_groups = organization_info.get_mapped_groups()
+            if (len(target_groups) == 0):
                 self.logger.info('No mapped groups for accessor: %s', organization_name) 
                 continue
                             
             for user_key in remove_user_key_list:
                 dashboard_user = organization_info.get_dashboard_user(user_key)
                 if (dashboard_user != None):
-                    groups_to_remove = self.normalize_groups(dashboard_user.get('groups')) & mapped_groups
+                    groups_to_remove = set()
+                    dashboard_group_names = self.normalize_groups(self.dashboard_user.get('groups'))
+                    for dashboard_group_name in dashboard_group_names:
+                        for target_group in target_groups:
+                            if target_group.group_name == dashboard_group_name:
+                                groups_to_remove.add(target_group)
+                                break
                 elif not organization_info.is_dashboard_users_loaded():
-                    groups_to_remove = mapped_groups
+                    groups_to_remove = target_groups
                 else:
                     groups_to_remove = None
 
                 if (groups_to_remove != None and len(groups_to_remove) > 0):
                     self.logger.info('Removing groups for user key: %s removed: %s', user_key, groups_to_remove)
                     username, domain = self.parse_user_key(user_key)
                     commands = user_sync.connector.dashboard.Commands(username=username, domain=domain)
-                    commands.remove_groups(groups_to_remove)
+
```

Comment:
Should have a helper function to remove groups.    The function would group by type internally.